cassandra.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"log"
+	"time"
+
+	"github.com/gocql/gocql"
+)
+
+type cassandraBackend struct {
+	Cluster *gocql.ClusterConfig
+	Session *gocql.Session
+}
+
+func (b *cassandraBackend) createSession() {
+	log.Println("Starting cassandra session")
+	session, err := b.Cluster.CreateSession()
+	if err != nil {
+		log.Fatal("Cassandra init failed")
+	}
+	b.Session = session
+}
+
+func (b *cassandraBackend) EndSession() {
+	log.Println("Ending cassandra session")
+	if b.Session != nil {
+		b.Session.Close()
+		b.Session = nil
+	}
+}
+
+// cassandraBackend must be made with this function
+func NewCassandraBackend(hosts []string) *cassandraBackend {
+	log.Println("Initializing cassandra backend")
+	log.Printf("Cassandra Hosts: %v\n", hosts)
+	b := new(cassandraBackend)
+	cluster := gocql.NewCluster(hosts...)
+	cluster.Keyspace = "catchall"
+	b.Cluster = cluster
+	b.createSession()
+	return b
+}
+
+type aggregatedResponse struct {
+	Count  int64
+	Bounce bool
+}
+
+func (b cassandraBackend) update() ([]HostEntry, error) {
+	var results []HostEntry
+	query := "SELECT host, lastseen from hosts"
```

Comment:
Same question, what if there's 10,000,000 records?

Reply:
Well in this situation the hosts list are a list of worker nodes (processes) that have run. Also they have a time which can be removed when stale.

The other table is a list of domains but 1 row per domain. 10,000,000 records would need to be discussed as to whether or not some of that data is meaningful. A time could be added as a column to locate stale data.

Reply:
Also I believe cassandra supports some style of sharding. I believe sharding would be a good solution for when it just gets big-big.

Reply:
I was thinking about adding some tests but I wanted to allow the PR to stabilize so people could get comments in. Though if that is not a problem for anyone I will do so.

Reply:
Yeah, makes sense. By default Cassandra limits the output and returns only 10,000 records :)

Reply:
I did not know that value. I assumed it existed but hadn't run into it yet. Thank you for the info :D